pangeo_forge_recipes/recipes/xarray_zarr.py

```diff
@@ -621,9 +621,25 @@ def store_chunk(
                 zarr_array[zarr_region] = data
 
 
-def finalize_target(target: CacheFSSpecTarget, consolidate_zarr: bool) -> None:
+def finalize_target(
+    target: CacheFSSpecTarget,
+    consolidate_zarr: bool,
+    consolidate_dimension_coordinates: bool = True,
+) -> None:
     if target is None:
         raise ValueError("target has not been set.")
+
+    if consolidate_dimension_coordinates:
+        logger.info("Consolidating dimension coordinate arrays")
+        target_mapper = target.get_mapper()
+        ds = xr.open_zarr(target_mapper)  # Probably a better way to get the dimension coords?
+        group = zarr.open(target_mapper)
+        for dim in ds.dims:
+            attrs = dict(group[dim].attrs)
+            data = group[dim][:]
```

Comment:
This assumes the coordinate fits in-memory on a single machine. I don't know if we're assuming that anywhere else (probably in the tests), but it's probably a safe assumption for now.

Reply:
I think it's a safe assumption, since the dimension coordinates are guaranteed to be 1D. We could note this caveat in the docs on `consolidate_dimension_coordinates`.

It also assumes that `dim in group`. But that's not always the case. Xarray Datasets can have dimensions with no corresponding coordinate, which would give a `KeyError` here.